torch/jit/__init__.py

```diff
@@ -631,10 +634,31 @@ def __getattr__(self, attr):
         return self.module._get_method(attr)
 
 
-def script(fn, optimize=True, _frames_up=0):
+def weak_script(fn, _frames_up=0):
+    compiled_weak_fns[fn] = {
+        "status": COMPILATION_PENDING,
+        "compiled_fn": None,
+        "rcb": createResolutionCallback(_frames_up + 1)
+    }
+    return fn
+
+
+def _try_compile_weak_script(fn):
+    entry = compiled_weak_fns.get(fn)
+    if entry is None:
+        return None
+    if entry["status"] == COMPILATION_PENDING:
+        entry["status"] = COMPILED
+        compiled_fn = torch.jit.script(fn, True, 0, entry["rcb"])
+        compiled_weak_fns[fn]["compiled_fn"] = compiled_fn
+    return compiled_fn
```

Comment:
Won't this be undefined on a second call with a function that should be compiled? It's only set in the block above